File: src/contests/cg/puzzle/php/the-fastest.php
<?php
// 🎮 CodinGame Puzzle - the-fastest
// https://www.codingame.com/training/medium/the-fastest

// Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
$n = (int) trim(fgets(STDIN));
$best = "";
for ($i = 0; $i < $n; $i++) {
    $t = trim(fgets(STDIN));
    if ($best === "" || $t < $best) {
        $best = $t;
    }
}
echo $best . "\n";

File: src/contests/cg/puzzle/dart/the-fastest.dart
// 🎮 CodinGame Puzzle - the-fastest
// https://www.codingame.com/training/medium/the-fastest

import 'dart:io';

void main() {
  // Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
  int n = int.parse(stdin.readLineSync()!.trim());
  String best = "";
  for (int i = 0; i < n; i++) {
    String t = stdin.readLineSync()!.trim();
    if (best == "" || t.compareTo(best) < 0) best = t;
  }
  print(best);
}
